core/bignum/div_spec.rb

```diff
@@ -4,11 +4,14 @@
 describe "Bignum#div" do
   it_behaves_like(:bignum_divide, :div)
 
-  it "returns a result of integer division of self by a float argument" do
-    bignum_value(88).div(0xffff_ffff.to_f).should eql(2147483648)
-    bignum_value(88).div(bignum_value(88).to_f).should eql(1)
-    bignum_value(88).div(-bignum_value(88).to_f).should eql(-1)
-  end
+    it "returns a result of integer division of self by a float argument" do
+      bignum_value(88).div(4294967295.5).should eql(2147483648)
+      not_supported_on :opal do 
+        bignum_value(88).div(4294967295.0).should eql(2147483648)
+        bignum_value(88).div(bignum_value(88).to_f).should eql(1)
+        bignum_value(88).div(-bignum_value(88).to_f).should eql(-1)
```

Comment:
I think this would be a littel bit confusing why Bignum#to_f returns a Bignum, but you are right the only thing that is canged by Bignum#to_f is that precision is getting lost. A "real" Fixnum with decimal places is not returned.